microcosm_pubsub/producer.py

```diff
@@ -100,16 +110,32 @@ def publish_message(self, pubsub_message: PubsubMessage):
             **pubsub_message.opaque_data
         )
         self.logger.debug("Publishing message with media type {media_type}", extra=extra)
+        publish_result = "SUCCESS"
+        publish_exception = None
 
         with elapsed_time(extra):
-            result = self.sns_client.publish(
-                TopicArn=pubsub_message.topic_arn,
-                Message=pubsub_message.message,
-                MessageAttributes=pubsub_message.message_attributes,
-            )
+            try:
+                result = self.sns_client.publish(
+                    TopicArn=pubsub_message.topic_arn,
+                    Message=pubsub_message.message,
+                    MessageAttributes=pubsub_message.message_attributes,
+                )
+            except Exception as e:
+                publish_exception = e
+                publish_result = "FAILURE"
+
+        self.pubsub_producer_metrics(
+            publish_result=publish_result,
+            **extra
+        )
+
+        if publish_exception:
+            raise Exception(f"Could not publish message, SNS producer error")
```

Comment:
Why not just re-raise the error?

Reply:
No reason outside of logging search

Reply:
Then why not:
```
             raise Exception(f"Could not publish message, SNS producer error: {publish_exception}")
```
I for one would like to know what the actual problem was.

Reply:
Done!